### PR TITLE
Feature/volunteer donate flow

### PR DESCRIPTION
**Pull request by Shoebmalik007**
Merge request includes:

1. Added carousels for volunteers and donations
2. fixed navigations of intro slider

### CODE REVIEW

File: src/Components/IntroSlider.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ const IntroSlider = () => {
     <View>
       <Button
         title="Skip"
-        onPress={() => navigation.navigate("LoginScreen")}
+        onPress={() => navigation.navigate("VolunteerHomeScreen")}
```
Comment (Megha-Alamance) [inline src/Components/IntroSlider.tsx:95]: @Shoebmalik007 , why is the Login screen removed from navigation and replaced with VolunteerHomeScreen? is it the flow and will it be same in v1 or it is just for v2

File: src/Screens/DonationTabScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [donationData, setDonationData] = useState<[]>([]);
+
+  const dispatch = useDispatch();
+
+  const navigation: any = useNavigation();
+
+  const fetchingDonationData = async () => {
+    const response = await dispatch(myDonations({} as any) as any);
+    console.log("checking repsonse from my events api", response);
```
Comment (Megha-Alamance) [inline src/Screens/DonationTabScreen.tsx]: @Shoebmalik007 , please remove console statements

File: src/Screens/DonationTabScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+  const handleSingleIndexSelect = async (index: any) => {
+    setSelectedIndex(index);
+    if (index === 0) {
+      fetchingDonationData();
+    } else if (index === 1) {
+      const res = await dispatch(allDonations({} as any) as any);
+      const donationsAll = res?.payload?.donationList;
+      const verifiedDonations = donationsAll?.filter(
+        (event: any) => event.status === "approved"
```
Comment (Megha-Alamance) [inline src/Screens/DonationTabScreen.tsx]: @Shoebmalik007 , please add null checks

File: src/Screens/DonationTabScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+      );
+
+      console.log(
```
Comment (Megha-Alamance) [inline src/Screens/DonationTabScreen.tsx]: @Shoebmalik007 , please remove console logs

File: src/Screens/DonationTabScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+        </View>
+        <ScrollView style={{ flex: 1 }}>
+          <FlatList
+            data={donationData}
+            renderItem={({ item }: any) => (
+              <Item
+                additionalInfo={item?.additionalInfo}
+                name={item?.name}
+                address={item?.address?.fullAddress}
+                lat={item.address?.lat}
```
Comment (Megha-Alamance) [inline src/Screens/DonationTabScreen.tsx]: @Shoebmalik007 , please add Null check item?.address?.lat

File: src/Screens/DonationTabScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+                long={item.address?.lng}
```
Comment (Megha-Alamance) [inline src/Screens/DonationTabScreen.tsx]: @Shoebmalik007 , same as above comment

File: src/Screens/VolunteerAndDonateScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+    { id: "1", title: "Item ", description: "Description " },
+    { id: "2", title: "Item ", description: "Description " },
+    { id: "3", title: "Item ", description: "Description " },
+  ];
+
+  const SecondRoute = () => {
+    const renderItem = ({ item }: any) => (
+      <TouchableOpacity>
+        <View style={styles.itemStyle}>
+          <Text style={styles.itemTitle}>{item.title}</Text>
```
Comment (Megha-Alamance) [inline src/Screens/VolunteerAndDonateScreen.tsx]: @Shoebmalik007 , please add null check item?.title

File: src/Screens/VolunteerAndDonateScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+          <Text style={styles.itemDescription}>{item.description}</Text>
```
Comment (Megha-Alamance) [inline src/Screens/VolunteerAndDonateScreen.tsx]: @Shoebmalik007 ,  please add null check item?.description

File: src/Screens/VolunteerAndDonateScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+        </View>
+      </TouchableOpacity>
+    );
+
+    return (
+      <FlatList
+        data={dummyData}
+        renderItem={renderItem}
+        keyExtractor={(item) => item.id}
```
Comment (Megha-Alamance) [inline src/Screens/VolunteerAndDonateScreen.tsx]: @Shoebmalik007 , please add null check

File: src/Screens/VolunteerDonateScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+  ];
+
+  const renderItem = ({ item }: any) => {
+    return (
+      <View>
+        <View style={[styles.card, { backgroundColor: "white" }]}>
+          <TouchableOpacity>
+            <View>
+              <Image
+                source={item.image}
```
Comment (Megha-Alamance) [inline src/Screens/VolunteerDonateScreen.tsx]: @Shoebmalik007 , please add null check

File: src/Screens/VolunteerDonateScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+            </View>
+            <View style={styles.headerContainer}>
+              <Text
+                style={{
+                  color: "white",
+                  fontSize: h2dp(3),
+                  marginLeft: w2dp(5),
+                }}
+              >
+                {item.title}
```
Comment (Megha-Alamance) [inline src/Screens/VolunteerDonateScreen.tsx]: @Shoebmalik007 , please add null check

File: src/redux/actions/myDonations.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+export interface MyDonations {
+  token: string;
+}
+
+export const myDonations = createAsyncThunk<MyDonations, MyDonations>(
+  "myDonations",
+  async (_, thunkAPI: any) => {
+    try {
+        const token = thunkAPI.getState().auth.data.token;
```
Comment (Megha-Alamance) [inline src/redux/actions/myDonations.ts]: @Shoebmalik007 , please add null checks

File: src/redux/actions/allDonations.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+export interface AllDonations {
+  token: string;
+}
+
+
+export const allDonations = createAsyncThunk<AllDonations, AllDonations>(
+  "allDonations",
+  async (_, thunkAPI: any) => {
+    try {
+        const token = thunkAPI.getState().auth.data.token;
```
Comment (Megha-Alamance) [inline src/redux/actions/allDonations.ts]: @Shoebmalik007 , please add null check

File: src/Utils/APIUtils.ts [hunk 14]
```diff
@@ -1,7 +1,7 @@
 import axios from "axios";
 
-const api_endPoint = "https://api.climatehealers.com/"
-// const api_endPoint = "https://foodhealers.climatehealers.com/"
+// const api_endPoint = "https://api.climatehealers.com/"
+const api_endPoint = "https://foodhealers.climatehealers.com/"
```
Comment (Megha-Alamance) [inline src/Utils/APIUtils.ts]: @Shoebmalik007 , why is this url used is this prod url? are we deploying for prod?

Comment (Shoebmalik007) [reply src/Utils/APIUtils.ts]: It was just for checking that everything is working properly  in prod.

File: src/Components/IntroSlider.tsx (hunk 1 above)
Comment (Shoebmalik007) [reply src/Components/IntroSlider.tsx:95]: the loginScreen was just for checking whether the skip button is working peoperly or not.
the volunteerhomescreen was not even developed so just for checking added the loginscreen

**Pull request merged**